packages/contracts-bedrock/src/cannon/MIPS2.sol

```diff
@@ -0,0 +1,642 @@
+                return outputState();
+            } else if (syscall_no == sys.SYS_FUTEX) {
+                // args: a0 = addr, a1 = op, a2 = val, a3 = timeout
+                if (a1 == sys.FUTEX_WAIT_PRIVATE) {
+                    thread.futexAddr = a0;
+                    uint32 mem = MIPSMemory.readMem(
+                        state.memRoot, a0 & 0xFFffFFfc, MIPSMemory.memoryProofOffset(MEM_PROOF_OFFSET, 1)
+                    );
+                    if (mem != a2) {
+                        v0 = 0xFF_FF_FF_FF;
+                        v1 = sys.EAGAIN;
+                    } else {
+                        thread.futexVal = a2;
+                        thread.futexTimeoutStep = a3 == 0 ? sys.FUTEX_NO_TIMEOUT : state.step + sys.FUTEX_TIMEOUT_STEPS;
+                        // Leave cpu scalars as-is. This instruction will be completed by `onWaitComplete`
+                        updateCurrentThreadRoot();
+                        return outputState();
+                    }
+                } else if (a1 == sys.FUTEX_WAKE_PRIVATE) {
+                    // Trigger thread traversal starting from the left stack until we find one waiting on the wakeup
+                    // address
+                    state.wakeup = a0;
+                    // Don't indicate to the program that we've woken up a waiting thread, as there are no guarantees.
+                    // The woken up thread should indicate this in userspace.
+                    st.CpuScalars memory cpu0 = getCpuScalars(thread);
+                    sys.handleSyscallUpdates(cpu0, thread.registers, 0, /* v0 */ 0 /* v1 */ );
+                    setStateCpuScalars(thread, cpu0);
+                    preemptThread(state, thread);
+                    state.traverseRight = false;
+                    return outputState();
+                } else {
+                    v0 = 0xFF_FF_FF_FF;
+                    v1 = sys.EINVAL;
+                }
+            } else if (syscall_no == sys.SYS_SCHED_YIELD) {
+                st.CpuScalars memory cpu0 = getCpuScalars(thread);
+                sys.handleSyscallUpdates(cpu0, thread.registers, 0, /* v0 */ 0 /* v1 */ );
+                setStateCpuScalars(thread, cpu0);
+                preemptThread(state, thread);
+                return outputState();
+            } else if (syscall_no == sys.SYS_OPEN) {
+                v0 = 0xFF_FF_FF_FF;
+                v1 = sys.EBADF;
+            } else if (syscall_no == sys.SYS_NANOSLEEP) {
+                st.CpuScalars memory cpu0 = getCpuScalars(thread);
+                sys.handleSyscallUpdates(cpu0, thread.registers, 0, /* v0 */ 0 /* v1 */ );
+                setStateCpuScalars(thread, cpu0);
+                preemptThread(state, thread);
+                return outputState();
+            } else if (syscall_no == sys.SYS_CLOCK_GETTIME) {
+                // ignored
+            } else if (syscall_no == sys.SYS_GET_AFFINITY) {
+                // ignored
+            }
```

Comment:
Are we going to revert / panic if we process an unrecognized syscall?  If not, seems like there's no point in explicitly ignoring these.

Reply:
We should. Though for these specific syscalls, the behavior we want is to return a "zero value" and a null errrno. This is handled at the top as `v0` and `v1` are set accordingly. I'll add a comment detailing this.